src/pint/models/parameter.py

```diff
@@ -201,124 +235,41 @@ def units(self, unt):
-
-        Parameters
-        ----------
-        value : array_like or float_like
-
-        Probabilities are evaluated using the value attribute
-        """
-        if value is None:
-            return (
-                self.prior.pdf(self.value)
-                if not logpdf
-                else self.prior.logpdf(self.value)
-            )
-        else:
-            return self.prior.pdf(value) if not logpdf else self.prior.logpdf(value)
-
-    # Setup .value property
-    # .value will get pure number from ._quantity.
-    # Setting .value property will change ._quantity.
-    @property
-    def value(self):
-        """Return the pure value of a parameter.
-
-        This value will associate with parameter default value, which is .units attribute.
-        """
-        if self._quantity is None:
-            return None
-        else:
-            return self.get_value(self._quantity)
-
-    @value.setter
-    def value(self, val):
-        """Set .value.
-
-        Setting .value attribute will change the .quantity attribute other than .value attribute.
-        """
-        if val is None:
-            if (
-                not isinstance(self.quantity, (str, bool))
-                and self._quantity is not None
-            ):
-                raise ValueError(
-                    "This parameter value is number convertible. "
-                    "Setting .value to None will lost the "
-                    "parameter value."
-                )
-            else:
-                self.value = val
-        self._quantity = self.set_quantity(val)
-
     @property
     def uncertainty(self):
-        """Return the internal stored parameter uncertainty value and units."""
+        """Parameter uncertainty value with units."""
         return self._uncertainty
 
     @uncertainty.setter
     def uncertainty(self, val):
-        """General wrapper setter for uncertainty.
-
-        The setting method is stored at .set_uncertainty attribute
-        """
         if val is None:
             if hasattr(self, "uncertainty") and self.uncertainty is not None:
-                raise ValueError(
-                    "Setting an exist uncertainty to None is not" " allowed."
-                )
+                raise ValueError("Setting an exist uncertainty to None is not allowed.")
             else:
-                self._uncertainty = val
-                self._uncertainty_value = self._uncertainty
+                self._uncertainty = self._uncertainty_value = None
                 return
-        self._uncertainty = self.set_uncertainty(val)
+        val = self._set_uncertainty(val)
 
-        # This is avoiding negtive unvertainty input.
-        if self._uncertainty is not None and self.uncertainty_value < 0:
-            self.uncertainty_value = np.abs(self.uncertainty_value)
+        if not val >= 0:
+            raise ValueError(f"Uncertainty cannot be set to negative {val}")
```

Comment:
Maybe"  "Uncertainties cannot be negative"?

Reply:
Clarified.